src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -286,6 +293,19 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             _reloadEnvironmentVariables = winrt::unbox_value_or<bool>(settings.TryLookup(L"reloadEnvironmentVariables").try_as<Windows::Foundation::IPropertyValue>(),
                                                                       _reloadEnvironmentVariables);
             _profileGuid = winrt::unbox_value_or<winrt::guid>(settings.TryLookup(L"profileGuid").try_as<Windows::Foundation::IPropertyValue>(), _profileGuid);
+
+            const auto& initialEnvironment{ winrt::unbox_value_or<winrt::hstring>(settings.TryLookup(L"initialEnvironment").try_as<Windows::Foundation::IPropertyValue>(), L"") };
+            if (!initialEnvironment.empty())
+            {
+                _initialEnv = til::env{ initialEnvironment.c_str() };
+            }
+            else
+            {
+                // If we were not explicitly provided an "initial" env block to
+                // treat as our original one, then just use our actual current
+                // env block.
+                _initialEnv = til::env::from_current_environment();
+            }
```

Comment:
After reading through the code I found one more nit I got a question about...

Here we stuff the environment eagerly into the `_initialEnv` just to then ignore it if `_reloadEnvironmentVariables` is true.

Why not check `_reloadEnvironmentVariables` right here and call `_initialEnv.regenerate()` as needed? That way we don't even need to store `_reloadEnvironmentVariables` as a member.